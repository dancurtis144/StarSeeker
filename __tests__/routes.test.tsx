import { render } from "@testing-library/react-native";

import RoutesScreen from "@/app/(tabs)/routes";

describe("RoutesScreen", () => {
  it("renders initial screen correctly", () => {
    const { getByText } = render(<RoutesScreen />);

    expect(getByText("Previous Routes")).toBeTruthy();
    expect(getByText("No previous routes available")).toBeTruthy();
  });

  it("matches snapshot", () => {
    const screen = render(<RoutesScreen />);
    expect(screen).toMatchSnapshot();
  });
});
