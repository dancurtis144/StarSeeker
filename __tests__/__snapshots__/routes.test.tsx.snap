// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutesScreen matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
        "paddingHorizontal": 32,
        "paddingTop": 80,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        {
          "fontSize": 32,
          "fontWeight": "bold",
          "lineHeight": 32,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Previous Routes
  </Text>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      {
        "gap": 20,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={10}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "flex": 1,
          "padding": 20,
        },
        {
          "backgroundColor": "#fff",
        },
      ]
    }
    viewabilityConfigCallbackPairs={[]}
    windowSize={2}
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              undefined,
              undefined,
              undefined,
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
              undefined,
              undefined,
            ]
          }
        >
          No previous routes available
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
